Makefile

```diff
@@ -175,11 +175,14 @@ build: buildsrc gen
 # get around a bug in go build where it will fail
 # to cache binaries from time to time on empty NFS
 # dirs
-buildsrc: crypto/libs/$(OS_TYPE)/$(ARCH)/lib/libsodium.a node_exporter NONGO_BIN deps $(ALGOD_API_SWAGGER_INJECT) $(KMD_API_SWAGGER_INJECT)
+buildsrc: check-go-version crypto/libs/$(OS_TYPE)/$(ARCH)/lib/libsodium.a node_exporter NONGO_BIN deps $(ALGOD_API_SWAGGER_INJECT) $(KMD_API_SWAGGER_INJECT)
```

Comment:
The default build target is `default`, which has the `build` target.
In order to make the `build`, it will try and follow the targets for `buildsrc` and `gen` in parallel. ( if -j is provided ).
But we want to make sure that before `gen` we have the ran the `check-go-version`, so you'll need to add that as a dependency for `gen` as well.

Plus, since it's not a real file, you'll need to add that to the `.PHONY` list.

Reply:
`gen` does not really matter and indirectly depends on `buildsrc` if I understand it right.
```
gen/%/genesis.json: gen/%.json gen/generate.go buildsrc
gen: $(addsuffix gen, $(NETWORKS)) mainnetgen
```